docarray/array/storage/base/seqlike.py

```diff
@@ -10,7 +11,15 @@ class BaseSequenceLikeMixin(MutableSequence[Document]):
     def _update_subindices_append_extend(self, value):
         if getattr(self, '_subindices', None):
             for selector, da in self._subindices.items():
-                docs_selector = DocumentArray(value)[selector]
+
+                value = DocumentArray(value)
+
+                if getattr(da, '_config', None) and da._config.root_id:
+                    for v in value:
+                        for doc in DocumentArray(v)[selector]:
+                            doc.tags['_root_id_'] = v.id
```

Comment:
The private key can't begin with `__` if we want to use `da[:, 'tags____sth__']` to get the attributes because [here](https://github.com/docarray/docarray/blob/main/docarray/helper.py#L86). So I changed `root_id` to `_root_id_`.